ipaserver/dns_data_management.py

```diff
@@ -443,6 +443,14 @@ def update_base_records(self):
             )
         )
 
+        # Remove the ipa-ca record(s). They will be reconstructed in
+        # get_base_records().
+        r_name = DNSName('ipa-ca') + self.domain_abs
+        try:
+            self.api_instance.Command.dnsrecord_del(
+                self.domain_abs, r_name, del_all=True)
+        except errors.NotFound:
+            pass
```

Comment:
Just a question:
can update of base dns records be done in a generic way without hardcoding specific values like `ipa-ca` (e.g. in `__update_dns_records`)?

Reply:
Only the ipa-ca record is affected by hidden masters AFAICT. The way the updates currently happen is for each A and AAAA record for each non-hidden host with a CA the ipa-ca record is constructed and written. The problem is if the last A or AAAA record for a CA is marked as hidden then the ipa-ca record for that record type will not be updated/removed, leaving the CA still visible.

I'll update 'ipa-ca' to ipalib.constants.IPA_CA_RECORD if that's what you're asking for.

I've been struggling to develop this test because I've been having problems with my local CI in running this particular suite for some reason, so I'm stuck trying to develop it on the public CI which is not optimal.

Reply:
Yes, this happens due to the implementation of `__update_dns_records` and its underlying `dnsrecord_mod`. If we want to ensure actual base DNS RRs (stored on DB) are synced to expected ones (e.g. output of `ipa dns-update-system-records`) then probably `dnsrecord_mod` is not suitable for this.
How about `del/add`?

Reply:
It's a fair point. It would prevent manual mistakes from hosing up the records, and pretty much everywhere else in IPA we do a del/add for similar reasons. I'll see how complex it is going to be.